### DISCUSSION TITLE
[FEATURE] Get the response from the bot as an array

### DISCUSSION THREAD

**Discussion by fenestron** [nutgram/nutgram#597] [Q&A]
### Is your feature request related to a problem? Please describe.

It would be great to be able to receive a response from the bot in an array format in tests (similar to how we can get a `json` response from a regular `POST` request)

### Describe the solution you'd like

An option to solve the problem is to add a new function to `FakeNutgram`:

```
    public function getResponseAsArray()
    {
        return array_map(function ($item) {
            [$request,] = array_values($item);
            return self::getActualData($request);
        }, $this->getRequestHistory());
    }
```

### Additional context

It would be convenient to receive the response as an array and use it in other requests to the bot or data assertions.

Example of use in Laravel tests:
```
<?php

namespace Tests\Feature;

use SergiX44\Nutgram\Nutgram;
use Tests\TestCase;

class NutgramTest extends TestCase
{
    public function test_feature_request(): void
    {
        $bot = app(Nutgram::class);
        $message = "new message";

        $answer = $bot
            ->hearText("$message")
            ->reply()
            ->getResponseAsArray();

        $this->assertEquals($message, $answer[0]['text']);
    }
}
```

**Comment by Lukasss93** [comment]
Why? Just do this:

```php
$bot
    ->hearText('/start')
    ->reply()
    ->assertReplyText('Hello world!'));
``` 

More informations: https://nutgram.dev/docs/testing/asserting
Example repository with tests: https://github.com/Lukasss93/telegram-stickeroptimizer/tree/master/tests/Feature

**Comment by fenestron** [comment]
> Why? Just do this:

What if I want to use the result of one message to send the next? For example:

```
public function test_feature_request(): void
    {
        $bot = app(Nutgram::class);
        $message = "new message";

        $answer = $bot
            ->hearText("$message")
            ->reply()
            ->getResponseAsArray();

        $bot->hearCallbackQueryData(
                $answer[0]['reply_markup']['inline_keyboard'][0][0]['callback_data']
            )
            ->reply();
    }
```

In this example, I send a message to the bot and receive an inline keyboard. I would like to be able to emulate pressing a button

**Comment by Lukasss93** [comment]
Hearing a callback query data should be in another test.
It should not be dynamic otherwise it would not be a real test.

```php
public function test_callback_query_data(): void
{
    $bot = app(Nutgram::class);

    $bot->hearCallbackQueryData('your-callback-data')
        ->reply();
        //assert...
}
``` 

If you want to test that the response has a button with a certain `callback_data` then use the `hearMessage` + `assertReplyMessage` methods.
